### PR TITLE
Some random cleanups and refactorings

### PR DESCRIPTION
**Pull request by mheinzel**
The accumulated over time on a branch.

### CODE REVIEW

File: services/brig/src/Brig/API.hs [hunk 1]
```diff
@@ -111,7 +111,7 @@ sitemap o = do
     accept "application" "json"
       .&. jsonRequest @NewUser
   put "/i/self/email" (continue changeSelfEmailNoSendH) $
-    header "Z-User"
+    zauthUserId
```
Comment (mheinzel) [inline services/brig/src/Brig/API.hs:114]: This slightly changes behaviour, as it masks the exact error message:

```
zheader = fmap (result (Fail . const accessDenied) Okay) . header
```

We already do the same in Galley, so it's probably a desirable change.

File: services/brig/src/Brig/User/API/Auth.hs [hunk 2]
```diff
@@ -260,13 +262,18 @@ renew ::
...
   Maybe (Either ZAuth.AccessToken ZAuth.LegalHoldAccessToken) ->
   Handler (Either (Auth.Access ZAuth.User) (Auth.Access ZAuth.LegalHoldUser))
-renew (Just (Left ut)) (Just (Left at)) = Left <$> Auth.renewAccess ut (Just at) !>> zauthError
-renew (Just (Left ut)) Nothing = Left <$> Auth.renewAccess @ZAuth.User @ZAuth.Access ut Nothing !>> zauthError
-renew (Just (Right lut)) (Just (Right lat)) = Right <$> Auth.renewAccess @ZAuth.LegalHoldUser @ZAuth.LegalHoldAccess lut (Just lat) !>> zauthError
-renew (Just (Right lut)) Nothing = Right <$> Auth.renewAccess @ZAuth.LegalHoldUser @ZAuth.LegalHoldAccess lut Nothing !>> zauthError
-renew (Just (Left _)) (Just (Right _)) = throwStd authTokenMismatch
-renew (Just (Right _)) (Just (Left _)) = throwStd authTokenMismatch
-renew Nothing _ = throwStd authMissingCookie
+renew = \case
```
Comment (mheinzel) [inline services/brig/src/Brig/User/API/Auth.hs:265]: Not sure if this really makes the code clearer, but I found it hard to follow the pattern matching and it was a nice challenge on a Friday afternoon. :)

File: services/spar/src/Spar/Scim.hs [hunk 3]
```diff
@@ -1,8 +1,8 @@
 {-# LANGUAGE DuplicateRecordFields #-}
 {-# LANGUAGE OverloadedLists #-}
 {-# LANGUAGE RecordWildCards #-}
-{-# OPTIONS_GHC -Wno-missing-methods
-    -Wno-orphans #-}
+{-# OPTIONS_GHC -Wno-missing-methods #-}
```
Comment (fisx) [inline services/spar/src/Spar/Scim.hs:4]: I wonder how that got in...?  :)

(I would have removed this and at least implemented the methods as `undefined`, but I guess that's not a great difference in terms of runtime behavior.)

File: libs/wai-utilities/src/Network/Wai/Utilities/Request.hs [hunk 4]
```diff
@@ -17,8 +18,7 @@
 -- You should have received a copy of the GNU Affero General Public License along
 -- with this program. If not, see <https://www.gnu.org/licenses/>.
 
-{-# OPTIONS -Wno-orphans #-}
--- for "instance HasRequest Request" :(
+-- orphan instance for "instance HasRequest Request" :(
```
Comment (fisx) [inline libs/wai-utilities/src/Network/Wai/Utilities/Request.hs:21]: This should also be moved up.

File: services/brig/src/Brig/User/API/Auth.hs (hunk 2 above)
Comment (fisx) [reply services/brig/src/Brig/User/API/Auth.hs:265]: Perhaps you can write 2-3 lines about what this does in English?  Should be a nice challenge, too.  :)P

File: libs/wai-utilities/src/Network/Wai/Utilities/Request.hs (hunk 4 above)
Comment (mheinzel) [reply libs/wai-utilities/src/Network/Wai/Utilities/Request.hs:21]: It can't, with current Ormolu. It will move all comments below the `OPTIONS_GHC` and `LANGUAGE` pragmas.
This was changed in Ormolu recently, but is not released yet: https://github.com/tweag/ormolu/blob/master/CHANGELOG.md

File: services/spar/src/Spar/Scim.hs (hunk 3 above)
Comment (mheinzel) [reply services/spar/src/Spar/Scim.hs:4]: And why is the instance needed, but not defining any of its methods doesn't break? How has this been in prod for two years? :grimacing: 

Looking into it now.

File: services/brig/src/Brig/User/API/Auth.hs (hunk 2 above)
Comment (mheinzel) [reply services/brig/src/Brig/User/API/Auth.hs:270]: I added some comments now.

But my this change didn't make it easier to understand the code, maybe I should leave it at the old version, which was verbose, but more explicit, and just add comments to that?

File: services/spar/src/Spar/Scim.hs (hunk 3 above)
Comment (mheinzel) [reply services/spar/src/Spar/Scim.hs:4]: So it seems like we host the full Scim `Site`, including the `GroupSite`, but the implementation for that goes through the `GroupDB` class, which would fail any time one of these endpoints is called.

I just made a commit to only host the `UserSite` and `ConfigSite` by copying a bit of code from `Web.Scim.Server`. This means people accessing group endpoints would get a proper 404 without hitting undefined methods. How does that sound to you?

About to push that in a moment.

File: services/brig/src/Brig/User/API/Auth.hs (hunk 2 above)
Comment (fisx) [reply services/brig/src/Brig/User/API/Auth.hs:270]: no, please keep it, now that it's not quite as late in the day and there is english, it's pretty!
